### PR TITLE
[MMSIG-80] Update and refine get_flops.py

### PR DESCRIPTION
**Pull request by xin-li-67**
## Motivation

MMSIG-80: update and refine get_flops.py

## Modification

tools/analysis_tools/get_flops.py

## Checklist

**Before PR**:

- [x] I have read and followed the workflow indicated in the [CONTRIBUTING.md](https://github.com/open-mmlab/mmpose/blob/master/.github/CONTRIBUTING.md) to create this PR.
- [x] Pre-commit or linting tools indicated in [CONTRIBUTING.md](https://github.com/open-mmlab/mmpose/blob/master/.github/CONTRIBUTING.md) are used to fix the potential lint issues.
- [x] Bug fixes are covered by unit tests, the case that causes the bug should be added in the unit tests.
- [x] New functionalities are covered by complete unit tests. If not, please add more unit tests to ensure correctness.
- [x] The documentation has been modified accordingly, including docstring or example tutorials.

**After PR**:

- [x] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

File: tools/analysis_tools/get_flops.py [hunk 1]
```diff
@@ -29,13 +32,13 @@ def parse_args():
...
         type=int,
         nargs='+',
         default=[256, 192],
         help='input image size')
     parser.add_argument(
-        '--input-constructor',
+        '--batch-input',
         '-c',
         type=str,
         choices=['none', 'batch'],
```
Comment (Ben-Louis) [inline tools/analysis_tools/get_flops.py:44]: Perhaps the argument can be converted to a boolean type using the `store_true` action? Additionally, the option '-c' may no longer be intuitive, as the argument name has been altered.

Comment (xin-li-67) [reply tools/analysis_tools/get_flops.py:44]: > Perhaps the argument can be converted to a boolean type using the `store_true` action? Additionally, the option '-c' may no longer be intuitive, as the argument name has been altered.

agree! I will update this in the next commit

**Comment by Tau-J** [comment]
1. `--batch-input` and `--batch-size` seems to be cumbersome. How about use `batch_input = args.batch_size > 1`?
2. I found a bug in mmengine: when `inputs=None`, `get_model_complexity_info()` cannot inference with `cuda:0`. I suggest set `cpu`  as default device for the moment. I'll report this bug to mmengine, or maybe someone will raise a PR to fix it.
3. The message is too long when printing both `out_arch` and `out_table`. I suggest only printing `out_table` and leaving `out_arch` as an optional argument.

**Comment by xin-li-67** [comment]
> 1. `--batch-input` and `--batch-size` seems to be cumbersome. How about use `batch_input = args.batch_size > 1`?
> 2. I found a bug in mmengine: when `inputs=None`, `get_model_complexity_info()` cannot inference with `cuda:0`. I suggest setting `cpu`  as default device for the moment. I'll report this bug to mmengine, or maybe someone will raise a PR to fix it.
> 3. The message is too long when printing both `out_arch` and `out_table`. I suggest only printing `out_table` and leaving `out_arch` as an optional argument.

I made some changes in the latest commit:
1. removed `batch-input` and use `batch-size`;
2. now default device is CPU;
3. now `out-table` will be printed by default, and `out-arch` will not be printed unless the user specifies with `--show-arch-info`. However, it seems that the `get_complexity_info` function in MMEngine still needs to be refined, some models in MMPose are not well supported.

**Comment by Tau-J** [comment]
LGTM

**Review by Tau-J — APPROVED** [review]

**Pull request merged**